hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -288,14 +302,17 @@ public BucketCache(String ioEngineName, long capacity, int blockSize, int[] buck
     this.ramCache = new ConcurrentHashMap<BlockCacheKey, RAMQueueEntry>();
 
     this.backingMap = new ConcurrentHashMap<BlockCacheKey, BucketEntry>((int) blockNumCapacity);
-
-    if (ioEngine.isPersistent() && persistencePath != null) {
+    if (ioEngine.isPersistent()) {
       try {
-        retrieveFromFile(bucketSizes);
+        if (persistencePath != null) {
+          retrieveFromFile(bucketSizes);
+        } else {
+          ((PersistentIOEngine) ioEngine).deleteCacheDataFile();
```

Comment:
If we delete data file here not, there wont be any functional diff I believe.  We were not doing it before. Any reason to do so now?  Or we do this delete in master branch now? Not remembering exactly.

Reply:
Do not delete data file does not affect the function, but it will occupy storage space.

Reply:
Data file can be large, since it is out-of-date and can't be restore, i'm fine with the deletion. Being kept is also good to me, only a matter of overwrite again.